src/pages/index.js

```diff
@@ -0,0 +1,361 @@
+
+// find the title input element
+const descriptionInputElement = editProfileModal.querySelector(
+  "#profile-description-input"
+);
+
+// Store card element and uniique card  id
+let selectedCard;
+let selectedCardId;
+
+//Edit button
+const cardModal = document.querySelector("#add-card-modal");
+// Setup another form element
+const cardForm = cardModal.querySelector(".modal__form");
+const cardSubmitBtn = cardModal.querySelector(".modal__submit-btn");
+const cardModalCloseBtn = cardModal.querySelector(".modal__close-btn");
+const profileAddButton = document.querySelector(".profile__add-button");
+
+//Select the modal delete button
+const deleteModalBtn = document.querySelector("#delete-modal");
+const deleteForm = deleteModalBtn.querySelector(".modal__form");
+
+//Select the avatar button
+// Avatar form elements
+const avatarModal = document.querySelector("#avatar-modal");
+const avatarForm = avatarModal.querySelector(".modal__form");
+const avatarSubmitBtn = avatarModal.querySelector(".modal__submit-btn");
+const avatarModalCloseBtn = avatarModal.querySelector(".modal__close-btn");
+const avatarAddButton = document.querySelector(".profile__avatar-edit-button");
+
+const cardLinkInput = cardModal.querySelector("#add-card-link-input");
+const cardNameInput = cardModal.querySelector("#add-card-name-input");
+
+//Select the modal
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImageEl = previewModal.querySelector(".modal__image");
+const previewModalCaption = previewModal.querySelector(".modal__caption");
+const previewCloseBtn = previewModal.querySelector(".modal__close-btn");
+
+//To set the template element
+const cardTemplate = document.querySelector("#card-template");
+// console.log(cardTemplate)
+
+//Create a function to call the card list
+const cardsList = document.querySelector(".cards__list");
+
+//Functions
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  const cardNameEL = cardElement.querySelector(".card__title");
+
+  //Select the image class and it's elements
+  const cardImage = cardElement.querySelector(".card__image");
+
+  //select the card like button element
+  const cardLikedBtn = cardElement.querySelector(".card__like-button");
+  const deleteCardBtn = cardElement.querySelector(".card__delete-button");
+
+  //assign  value to the name
+  cardNameEL.textContent = data.name;
+  //check if the card is liked or not
+
+  //assign  values to the image src and alt
+  cardImage.src = data.link;
+  cardImage.alt = data.name;
+
+  // Safely retrieve and parse liked cards from localStorage
+  let likedCards = {};
+
+  try {
+    likedCards = JSON.parse(localStorage.getItem("likedCards")) || {};
+  } catch (error) {
+    console.error("Error parsing likedCards from localStorage:", error);
+    localStorage.setItem("likedCards", JSON.stringify({})); // Reset storage
```

Comment:
`localStorage` can't be used for **liking**. Everything should be checked according to the task